common/include/pcl/pcl_macros.h

```diff
@@ -76,6 +76,36 @@
 
 #include <pcl/pcl_config.h>
 
+#define PCL_SINGLE_ARG(...) __VA_ARGS__
+
+#if (defined(__CUDACC__) && defined(_MSC_VER))
+  // No version of the 'deprecated' attribute works when
+  // compiling with nvcc under Windows until atleast CUDA 10.2.
+  // When this is fixed we can correctly define the attribute.
+  #define PCL_DEPRECATED
+  #define PCL_DEPRECATED_MESSAGE(message)
+  #define PCL_DEPRECATED_USING(identifier, type, message) using identifier = type
+#elif (__cplusplus >= 201402L || (defined(_MSC_VER) && (_MSC_VER >= 1910) && (_MSVC_LANG >= 201402L)))
```

Comment:
This can be better expressed as 
```c
#if defined(__has_cpp_attribute) && __has_cpp_attribute(deprecated)
...
#elif defined(__GNUC__) || defined(__clang__)  // both have same compiler intrinsic
...
#elif defined(_MSC_VER)
...
#else
#warning
...
#endif
```